runtime/sema/checker.go

```diff
@@ -122,34 +122,16 @@ type Checker struct {
 
 type Option func(*Checker) error
 
-func WithPredeclaredValues(predeclaredValues []ValueDeclaration) Option {
+func WithBaseValueActivation(baseValueActivation *VariableActivation) Option {
 	return func(checker *Checker) error {
-		checker.PredeclaredValues = predeclaredValues
-
-		for _, declaration := range predeclaredValues {
-			variable := checker.declareValue(declaration)
-			if variable == nil {
-				continue
-			}
-			checker.Elaboration.GlobalValues.Set(variable.Identifier, variable)
-			checker.Elaboration.EffectivePredeclaredValues[variable.Identifier] = declaration
-		}
-
+		checker.baseValueActivation = baseValueActivation
 		return nil
```

Comment:
This is one of the main optimizations: Instead of re-declaring the predeclared values (and types below; same for the predeclared values in the interpreter), a fully prepared activation is linked into the variable activations stack

Reply:
nice! That makes total sense